eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubClientBuilder.java

```diff
@@ -95,15 +139,50 @@ public EventHubClientBuilder retry(RetryPolicy retryPolicy) {
     }
 
     /**
-     * Creates a new {@link EventHubClient} based on the configuration set in this builder.
+     * Sets the configuration store that is used during construction of the service client.
      *
-     * @return A new {@link EventHubClient} instance.
+     * The default configuration store is a clone of the {@link ConfigurationManager#getConfiguration() global
+     * configuration store}, use {@link Configuration#NONE} to bypass using configuration settings during construction.
+     *
+     * @param configuration The configuration store used to
+     * @return The updated EventHubClientBuilder object.
      */
-    public EventHubClient build() {
-        return new EventHubClient();
+    public EventHubClientBuilder configuration(Configuration configuration) {
+        this.configuration = configuration;
+        return this;
     }
 
-    //TODO (conniey): Remove placeholder when the client is updated.
-    public static class ProxyConfiguration {
+    private ProxyConfiguration constructDefaultProxyConfiguration(Configuration configuration) {
+        ProxyAuthenticationType authentication = ProxyAuthenticationType.NONE;
+        if (this.proxyConfiguration != null) {
+            authentication = this.proxyConfiguration.authentication();
+        }
+
+        String proxyAddress = configuration.get(BaseConfigurations.HTTP_PROXY);
+        String host;
+        Integer port;
+        Proxy proxy = null;
+        if (proxyAddress != null) {
+            String[] hostPort = proxyAddress.split(":");
+            if (hostPort.length < 2) {
+                throw new IllegalArgumentException("HTTP_PROXY cannot be parsed into a proxy");
+            }
+            host = hostPort[0];
+            port = Integer.parseInt(hostPort[1]);
+            proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(host, port));
+        }
+
+        String username = configuration.get(ProxyConfiguration.PROXY_USERNAME);
+        String password = configuration.get(ProxyConfiguration.PROXY_PASSWORD);
+        if (authentication == ProxyAuthenticationType.BASIC || authentication == ProxyAuthenticationType.DIGEST) {
```

Comment:
This if check can be removed. in the case that They want to use BASIC or DIGEST and do not pass in a username or password, the WebProxyHandler will fetch the System Defaults.